internals/webhooks/defaults_webhook_test.go

```diff
@@ -0,0 +1,74 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	rcsv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	. "github.com/onsi/gomega"
+	k8sadm "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	knativev1 "knative.dev/serving/pkg/apis/serving/v1"
+	testclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type TestCase struct {
+	cappName    string
+	scaleMetric string
+	mutated     bool
+}
+
+func TestDefaultsWebhook(t *testing.T) {
+	tests := []TestCase{
+		{
+			cappName: "i-forgot-metric",
+			mutated:  true,
+		}, {
+			cappName:    "i-put-metric",
+			scaleMetric: "cpu",
+			mutated:     false,
+		},
+	}
+	client := testclient.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects().Build()
+	for _, tc := range tests {
+		t.Run(tc.cappName, func(t *testing.T) {
+			ctx := context.Background()
+			capp := rcsv1alpha1.Capp{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: tc.cappName,
+				},
+				Spec: rcsv1alpha1.CappSpec{
+					ConfigurationSpec: knativev1.ConfigurationSpec{},
+					ScaleMetric:       tc.scaleMetric,
+					Site:              "local-cluster",
+					RouteSpec: rcsv1alpha1.RouteSpec{
+						Hostname: "test.com",
+					},
+				},
+			}
+			g := NewWithT(t)
+			decoder, _ := admission.NewDecoder(scheme.Scheme)
+			rc := DefaultMutator{Decoder: decoder, Client: client}
+			obj, err := json.Marshal(capp)
+			g.Expect(err).ShouldNot(HaveOccurred())
+
+			req := admission.Request{AdmissionRequest: k8sadm.AdmissionRequest{Name: capp.Name,
+				Kind:   metav1.GroupVersionKind{Kind: "Capp", Group: "rcs.dana.io", Version: "v1alpha1"},
+				Object: runtime.RawExtension{Raw: obj},
+			}}
+
+			response := rc.Handle(ctx, req)
```

Comment:
I don't think we need to create a `req`, I'd rather we check the `handle` function and not the `Handle` function, similar to how it's done here: https://github.com/dana-team/hostedcluster-requester/blob/main/internal/webhooks/hostedcluster_webhook_test.go